backend/FwLite/FwLiteDesktop/AppUpdateService.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Buffers;
+using System.Net.Http.Json;
+using Windows.Management.Deployment;
+using LexCore.Entities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Win32;
+
+namespace FwLiteDesktop;
+
+public class AppUpdateService(
+    IHttpClientFactory httpClientFactory,
+    ILogger<AppUpdateService> logger,
+    IPreferences preferences) : IMauiInitializeService
+{
+    private const string LastUpdateCheck = "lastUpdateChecked";
+    private const string FwliteUpdateUrlEnvVar = "FWLITE_UPDATE_URL";
+    private const string ForceUpdateCheckEnvVar = "FWLITE_FORCE_UPDATE_CHECK";
+    private const string PreventUpdateCheckEnvVar = "FWLITE_PREVENT_UPDATE";
+
+    private static readonly SearchValues<string> ValidPositiveEnvVarValues =
+        SearchValues.Create(["1", "true", "yes"], StringComparison.OrdinalIgnoreCase);
+
+    private static readonly string UpdateUrl = Environment.GetEnvironmentVariable(FwliteUpdateUrlEnvVar) ??
+                                               $"https://lexbox.org/api/fwlite-release/latest?appVersion={AppVersion.Version}";
+
+    public void Initialize(IServiceProvider services)
+    {
+        _ = Task.Run(TryUpdate);
+    }
+
+    private async Task TryUpdate()
+    {
+        if (ValidPositiveEnvVarValues.Contains(Environment.GetEnvironmentVariable(PreventUpdateCheckEnvVar) ?? ""))
+        {
+            logger.LogInformation("Update check prevented by env var {EnvVar}", PreventUpdateCheckEnvVar);
+            return;
+        }
+
+        if (!ShouldCheckForUpdate()) return;
+        var latestRelease = await FetchRelease();
+        if (latestRelease is null) return;
+        var currentVersion = AppVersion.Version;
+        var shouldUpdateToRelease = String.Compare(latestRelease.Version, currentVersion, StringComparison.Ordinal) > 0;
```

Comment:
This is what we've got currently, right:
`<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>`

Also, just so it's been said: if we were using the same version format we have for Lexbox (e.g. "v2024-11-27-7885c5df"), there'd be a 50% chance that a new release on the same day would be considered an older release, because the commit hashes aren't comparable.

Reply:
actually I don't think it was working correctly. App version pulls the `AssemblyInformationalVersionAttribute` which I wasn't setting on the MAUI project, it's being set now.

You are correct about the version matching for releases on the same day. I'm not sure if it's worth the effort to solve it though as it's rare, though I guess when we need it is when it will hurt the most to not have it. I suppose we could move the code to determine if we update to the server. So the client just sends it's version number and the server returns true or false to tell the client to update. That would let us resolve this issue in a number of different ways (eg detect this case and figure out which is latest from github release info which we already have)